src/cls/rgw/cls_rgw.cc

```diff
@@ -978,6 +1076,12 @@ static int complete_remove_obj(cls_method_context_t hctx,
           int(entry.meta.category));
   unaccount_entry(header, entry);
 
+  bufferlist empty;
+  ret = record_duplicate_entry(hctx, idx, key, &empty,
+                               header.resharding_in_logrecord());
+  if (ret < 0)
+    return ret;
+
```

Comment:
can replace this and the `cls_cxx_map_remove_key()` call just below with `remove_entry()`